src/Microsoft.ML.AutoML/Tuner/PipelineProposer.cs

```diff
@@ -83,16 +84,17 @@ public PipelineProposer(SweepablePipeline sweepablePipeline, AutoMLExperimentSet
             }
 
             _pipelineSchemas = _sweepablePipeline.Schema.ToTerms().Select(t => t.ToString()).ToArray();
+            _learnerInitialCost = _pipelineSchemas.ToDictionary(kv => kv, kv => GetEstimatedCostForPipeline(kv, _sweepablePipeline));
+            // initialize eci with the estimated cost and always start from pipeline which has lowest cost.
+            _eci = _pipelineSchemas.ToDictionary(kv => kv, kv => GetEstimatedCostForPipeline(kv, _sweepablePipeline));
         }
 
-        public (SearchSpace.SearchSpace, string) ProposeSearchSpace()
+        public string ProposeSearchSpace()
         {
-            _learnerInitialCost = _pipelineSchemas.ToDictionary(kv => kv, kv => GetEstimatedCostForPipeline(kv, _sweepablePipeline));
             string schema;
-            if (_eci == null)
+            if (!_initialized)
             {
-                // initialize eci with the estimated cost and always start from pipeline which has lowest cost.
-                _eci = _pipelineSchemas.ToDictionary(kv => kv, kv => GetEstimatedCostForPipeline(kv, _sweepablePipeline));
+                _initialized = true;
```

Comment:
>_initialized [](http://example.com/codeflow?start=16&length=12)

nit: move this line after setting the schema. just in case to be safer for multi-threading scenarios if it is possible.

Reply:
### Resolved